rollout.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+// Rollout is an interface implemented by a value that can provide a rollout
+// percentage as a decimal in the range [0.0,1.0]
+type Rollout interface {
+	// ReadRollout provides the current rollout value.
+	// Note that it is possible for the value to be outside of the
+	// acceptable [0.0,1.0] range - this should be checked by the caller.
+	ReadRollout() float64
+}
+
+// A DynamoDBRollout represents a rollout that is continuously fetched from
+// DynamoDB.
+// The DynamoDB table must have a hash string key of "application", a range
+// string key of "version", and a rollout number value stored under the key
+// "rollout".
+// "version" must always be set to "canary".
+// If enough calls to DynamoDB fail, the rollout value will drop to 0 to
+// minimize possible damange (i.e. the inability to rollback a canary).
+type DynamoDBRollout struct {
+	db      *dynamodb.DynamoDB
+	table   string
+	mutex   *sync.RWMutex
+	rollout float64
+}
+
+// NewDynamoDBRollout creates a new DynamoDBRollout and begins eternally
+// querying DynamoDB for canary values.
+func NewDynamoDBRollout(monitor Monitor, db *dynamodb.DynamoDB, table string, application string, delay time.Duration, unhealthy time.Duration) *DynamoDBRollout {
+	if db == nil {
+		return nil
```

Comment:
I think the convention is that the New* functions return (*Impl, error). In this case, I think you might want to 

return nil, fmt.Errorf("db can not be nil")

or something.